gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/drilling/Drill.java

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright © 2014 - 2017 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.drilling;
+
+import org.gradoop.flink.model.api.operators.UnaryGraphToGraphOperator;
+import org.gradoop.flink.model.impl.operators.drilling.functions.drillfunctions.DrillFunction;
+
+import java.util.Objects;
+
+/**
+ * Base class for the roll up / drill down operation. Also contains a builder for these.
+ */
+public abstract class Drill implements UnaryGraphToGraphOperator {
+
+  /**
+   * Used as new property key to declare that the current property key shall be kept.
+   */
+  public static final String KEEP_CURRENT_PROPERTY_KEY = ":keepKey";
+  /**
+   * Used as label to declare that all vertices or all edges shall be drilled.
+   */
+  public static final String DRILL_ALL_ELEMENTS = ":allElements";
+  /**
+   * Separator between the iteration number and the original property key when the property key
+   * shall be kept.
+   */
+  public static final String PROPERTY_VERSION_SEPARATOR = "__";
+
+  /**
+   * Label of the element whose property shall be drilled.
+   */
+  private String label;
+  /**
+   * Property key.
+   */
+  private String propertyKey;
+  /**
+   * Drill function which shall be applied to a property.
+   */
+  private DrillFunction function;
+  /**
+   * New property key.
+   */
+  private String newPropertyKey;
+  /**
+   * True, if vertices shall be drilled, false for edges
+   */
+  private boolean drillVertex;
+
+  /**
+   * Valued constructor.
+   *  @param label label of the element whose property shall be drilled, or
+   *              see {@link Drill#DRILL_ALL_ELEMENTS}
+   * @param propertyKey property key
+   * @param function drill function which shall be applied to a property
+   * @param newPropertyKey new property key, or see {@link Drill#KEEP_CURRENT_PROPERTY_KEY}
+   * @param drillVertex true, if vertices shall be drilled, false for edges
+   */
+  public Drill(String label, String propertyKey, DrillFunction function, String newPropertyKey,
```

Comment:
overload constructor without `String label` which defines the default case => all labels